src/invoices/Tags/TagsForm.js

```diff
@@ -0,0 +1,136 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import {
+  difference,
+  isEqual,
+  noop,
+  sortBy,
+} from 'lodash';
+
+import { MultiSelection } from '@folio/stripes/components';
+
+const TagsForm = ({
+  allTags,
+  entityTags,
+  onAdd,
+  onRemove,
+}) => {
+  const [savedTags, setSavedTags] = useState([]);
+
+  useEffect(() => {
+    if (!isEqual(entityTags, savedTags)) {
+      setSavedTags(entityTags);
+    }
+  }, [entityTags, savedTags]);
+
+  const onChange = useCallback(
+    (tags) => {
+      const updatedTags = tags.map(t => t.value);
+
+      if (tags.length < savedTags.length) {
+        const tag = difference(savedTags, tags.map(t => t.value));
+
+        onRemove(tag[0]);
+      } else onAdd(updatedTags);
+
+      setSavedTags(updatedTags);
+    },
+    [onAdd, onRemove, savedTags],
+  );
+
+  const filterItems = useCallback(
+    (filter, list) => {
+      if (!filter) {
+        return { renderedItems: list };
+      }
+
+      const filtered = list.filter(item => item.value.match(new RegExp(filter, 'i')));
+      const renderedItems = filtered.sort((tag1, tag2) => {
+        const regex = new RegExp(`^${filter}`, 'i');
+        const match1 = tag1.value.match(regex);
+        const match2 = tag2.value.match(regex);
+
+        if (match1) return -1;
+        if (match2) return 1;
+
+        return (tag1.value < tag2.value) ? -1 : 1;
+      });
+
+      return { renderedItems };
+    },
+    [],
+  );
+
+  const addTag = useCallback(
+    ({ inputValue }) => {
+      const tag = inputValue.replace(/\s|\|/g, '').toLowerCase();
+      const updatedTags = savedTags.concat(tag);
+
+      setSavedTags(updatedTags);
+      onAdd(updatedTags);
+    },
+    [savedTags, onAdd],
```

Comment:
according to https://reactjs.org/docs/hooks-reference.html#usecallback `useCallback` will not work here, since `onAdd` is a function and it always a new one. Please correct me if I'm wrong
cc @NikitaSedyx

Reply:
actually `onAdd` is wrapped by useCallback